### PR TITLE
Trigger Target Update From WTMULT in ACTIONX

### PR DESCRIPTION
**Pull request by bska**
We need to inform the simulator about possible changes to wells when running `WTMULT` in an `ACTIONX` block.  Recording the keyword's affected wells in the handler context enables the simulator to run any requisite post-action operations such as updating the active wells' operating targets.  Without the call to
```
HandlerContext::affected_well
```
those post-action operations might not run.

### CODE REVIEW

**Comment by bska** [comment]
I am creating this PR in draft mode for a few reasons

1. It depends on, and contains, the earlier PR #4093
2. It depends on the downstream PR OPM/opm-simulators#5411 to enable running models with the `WTMULT` keyword at all

I will keep the PR in a draft state until such time as it is ready for review and merging.  That said, when combined with OPM/opm-simulators#5411 and OPM/opm-simulators#5409, I am able to run the [`WTMULT-03`](https://github.com/OPM/opm-tests/blob/cc1f88b5c444368321e24122e5b008d57b32aee7/wtmult/WTMULT-03.DATA) test model (PR OPM/opm-tests#1172, uses `WTMULT` in an [`ACTIONX` block](https://github.com/OPM/opm-tests/blob/cc1f88b5c444368321e24122e5b008d57b32aee7/wtmult/WTMULT-03.DATA#L271-L280)) and get reasonably encouraging results.

**Comment by bska** [comment]
I've decided to pull this apart from the upstream PR #4093.  That work is still undergoing design discussions whereas this is a more clear-cut bug fix.  I'll mark the PR as "ready for review" and run a build check.

**Comment by bska** [comment]
jenkins build this please

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 1]
```diff
@@ -1091,22 +1091,23 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
...
-
-      [1]: The leading '*' in a WLIST name should not be interpreted as a shell
-           wildcard!
-    */
-
-
-    std::vector<std::string> Schedule::wellNames(const std::string& pattern, std::size_t timeStep, const std::vector<std::string>& matching_wells) const {
+    // There are many SCHEDULE keyword which take a wellname as argument.
+    // In addition to giving a fully qualified name like 'W1' you can also
+    // specify shell wildcard patterns like like 'W*', you can get all the
```
Comment (akva2) [inline opm/input/eclipse/Schedule/Schedule.cpp:1096]: like like

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 2]
```diff
@@ -1091,22 +1091,23 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
...
+    // wells in the well-list '*WL'[1] and the wellname '?' is used to get
+    // all the wells which already have matched a condition in a ACTIONX
```
Comment (akva2) [inline opm/input/eclipse/Schedule/Schedule.cpp:1098]: an ACTIONX

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 3]
```diff
@@ -1091,22 +1091,23 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
...
+    // keyword.  This function should be one-stop function to get all well
```
Comment (akva2) [inline opm/input/eclipse/Schedule/Schedule.cpp:1099]: an one-stop

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 4]
```diff
@@ -1091,22 +1091,23 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
...
+    // names according to a input pattern.  The timestep argument is used to
```
Comment (akva2) [inline opm/input/eclipse/Schedule/Schedule.cpp:1100]: an input

File: opm/input/eclipse/Schedule/Action/ActionX.cpp [hunk 5]
```diff
@@ -15,25 +15,38 @@
```
Comment (akva2) [inline opm/input/eclipse/Schedule/Action/ActionX.cpp:15]: (can't comment on commit message). corder -> order

**Review by akva2 — APPROVED** [review]
Some spelling stuff that we may as well fix while touching the lines.

**Comment by bska** [comment]
> Some spelling stuff that we may as well fix while touching the lines.

Thank you for the comments.  I **believe** I've addressed those in the most recent update here, but do feel free to have a second look.  In the meantime I'll rerun the build check.

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
> jenkins build this please

PR approved and build check is green.  I'll merge into master.

**Pull request merged**